cuqi/geometry/_geometry.py

```diff
@@ -650,17 +651,27 @@ class KLExpansion(Continuous1D):
 
     normalizer : float, default 1.0
         A factor of the basis functions shown in the formula above.
+
+    num_modes : int, default None
+        Number of expansion modes to use in the KL expansion. If `num_modes` is 
+        None or larger than the number of grid points, all modes will be used.
+
     """
-    
+
     # init function defining parameters for the KL expansion
-    def __init__(self, grid,  decay_rate=2.5, normalizer=12.0, axis_labels=None, **kwargs):
+    def __init__(self, grid=None, decay_rate=2.5, normalizer=12.0, num_modes=None, axis_labels=None, **kwargs):
```

Comment:
Could you add doc to explain what happens when grid = None?

Reply:
I updated the documentation for grid.